pyrobosim/pyrobosim/core/world.py

```diff
@@ -1794,13 +1824,21 @@ def is_path_collision_free(self, path: Path, step_dist: float = 0.01) -> bool:
         """
         for idx in range(len(path.poses) - 1):
             if not self.is_connectable(
-                path.poses[idx], path.poses[idx + 1], step_dist=step_dist
+                path.poses[idx],
+                path.poses[idx + 1],
+                step_dist=step_dist,
+                partial_observability_hallway_states=partial_observability_hallway_states,
+                recorded_closed_hallways=recorded_closed_hallways,
             ):
                 return False
         return True
 
     def sample_free_robot_pose_uniform(
-        self, robot: Robot | None = None, ignore_robots: bool = True
+        self,
+        robot: Robot | None = None,
+        ignore_robots: bool = True,
+        partial_observability_hallway_states: bool = False,
```

Comment:
I think you can just grab this and the next argument from the `robot` object in this case, rather than making new arguments, right?